### PR TITLE
feat: Write logged features to Offline Store (Go - Python integration)

### PR DESCRIPTION
**Pull request by pyalex**
Signed-off-by: pyalex <moskalenko.alexey@gmail.com>

**What this PR does / why we need it**:

Following previous PRs (#2574 and #2614) this PR finalizes integration between producing logs in Go feature server and writing them to an offline store via python implementation.
Here I added end-to-end test that starts go gRPC server and check that offline store has received served features.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: go/embedded/online_features.go [hunk 1]
```diff
@@ -215,16 +223,44 @@ func (s *OnlineFeatureService) GetOnlineFeatures(
...
+
+func (s *OnlineFeatureService) StartGprcServerWithLoggingDefaultOpts(host string, port int, writeLoggedFeaturesCallback logging.OfflineStoreWriteCallback) error {
+	defaultOpts := LoggingOptions{
+		ChannelCapacity: logging.DefaultOptions.ChannelCapacity,
+		EmitTimeout:     logging.DefaultOptions.EmitTimeout,
+		WriteInterval:   logging.DefaultOptions.WriteInterval,
+		FlushInterval:   logging.DefaultOptions.FlushInterval,
+	}
+	return s.StartGprcServerWithLogging(host, port, writeLoggedFeaturesCallback, defaultOpts)
+}
```
Comment (achals) [inline go/embedded/online_features.go:237]: What's the difference between default logging opts and no logging opts? Is logging disable? Can we add comments for the methods?

File: go/embedded/online_features.go [hunk 2]
```diff
@@ -234,6 +270,10 @@ func (s *OnlineFeatureService) StartGprcServer(host string, port int) error {
 		<-s.grpcStopCh
 		fmt.Println("Stopping the gRPC server...")
 		grpcServer.GracefulStop()
+		if loggingService != nil {
+			loggingService.Stop()
+		}
+		fmt.Println("gRPC server terminated")
```
Comment (achals) [inline go/embedded/online_features.go:276]: We should really move some of this stuff to use a proper logger.

**Review by achals — APPROVED** [review]

File: go/internal/feast/server/logging/offlinestoresink.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	"path/filepath"
+
+	"github.com/apache/arrow/go/v8/arrow"
+	"github.com/apache/arrow/go/v8/arrow/array"
+	"github.com/apache/arrow/go/v8/parquet"
+	"github.com/apache/arrow/go/v8/parquet/pqarrow"
+	"github.com/google/uuid"
+)
+
+type OfflineStoreWriteCallback func(featureServiceName, datasetDir string) string
```
Comment (kevjumba) [inline go/internal/feast/server/logging/offlinestoresink.go:18]: probably should specify name as string too?

Comment (pyalex) [reply go/internal/feast/server/logging/offlinestoresink.go:18]: it's a short form

File: go/embedded/online_features.go (hunk 1 above)
Comment (pyalex) [reply go/embedded/online_features.go]: Added

**Pull request merged**